helix-core/src/main/java/org/apache/helix/manager/zk/ParticipantManager.java

```diff
@@ -362,6 +363,7 @@ private void carryOverPreviousCurrentState() {
         }
 
         // If the the current state is related to tasks, there is no need to carry it over to new session.
+        // Note: this check is not necessary due to TaskCurrentStates, but keep it for backwards compatibility
```

Comment:
I am not sure if I understand this comment. We have this check to not carry over task related current states. Why it is not necessary?

Reply:
Because this logic is working with `_keyBuilder.currentStates`. TaskCurrentStates have their own path now (`keyBuilder.taskCurrentStates`), meaning this check will not be used unless for backward compatibility purposes.